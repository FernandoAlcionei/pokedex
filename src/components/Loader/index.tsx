import { Loader2 } from "lucide-react";

const Loader = () => (
    <div className="absolute top-0 bottom-0 left-0 right-0 bg-[#00000042] flex items-center justify-center">
        <Loader2 color='#FFF' className="mr-2 h-14 w-14 animate-spin" />
    </div>
);

export default Loader;