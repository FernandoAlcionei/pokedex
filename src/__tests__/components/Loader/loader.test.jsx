import { render } from "@testing-library/react";
import Loader from "@/components/Loader";

describe("Loader Component", () => {
  it("should render the Loader component correctly", () => {
    const { container } = render(
      <Loader />
    );

    expect(container).toMatchSnapshot();
  });
});
