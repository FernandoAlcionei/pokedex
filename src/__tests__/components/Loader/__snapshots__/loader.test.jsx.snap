// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render correctly 1`] = `
<div>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 bg-[#00000042] flex items-center justify-center"
  >
    <svg
      class="lucide lucide-loader-circle mr-2 h-14 w-14 animate-spin"
      fill="none"
      height="24"
      stroke="#FFFFFF"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </div>
</div>
`;

exports[`Loader Component should render the Loader component correctly 1`] = `
<div>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 bg-[#00000042] flex items-center justify-center"
  >
    <svg
      class="lucide lucide-loader-circle mr-2 h-14 w-14 animate-spin"
      fill="none"
      height="24"
      stroke="#FFFFFF"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </div>
</div>
`;

exports[`Loader should render correctly 1`] = `
<div>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 bg-[#00000042] flex items-center justify-center"
  >
    <svg
      class="lucide lucide-loader-circle mr-2 h-14 w-14 animate-spin"
      fill="none"
      height="24"
      stroke="#FFFFFF"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </div>
</div>
`;
